### PR TITLE
Add CIS 1.4

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

 adds rule 1.4 of CIS GCP
Screenshot 2023-07-18 at 9 27 30

### Related Issues

- https://github.com/elastic/security-team/issues/6179

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_1_4/rule.rego [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	}
+}
+
+finding = result if {
+	data_adapter.is_iam
+	data_adapter.has_policy
+
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_valid),
+		{},
```
Comment (orouz) [inline bundle/compliance/cis_gcp/rules/cis_1_4/rule.rego:22]: TODO: add evidence

File: bundle/compliance/cis_gcp/test_data.rego [hunk 2]
```diff
@@ -45,3 +57,10 @@ not_eval_resource = {
 	"type": "key-management",
 	"subType": "no-exisitng-type",
 }
+
+# missing resource.iam_policy
+no_policy_resource = {
```
Comment (orouz) [inline bundle/compliance/cis_gcp/test_data.rego:62]: this is duplicated, will change to `generate_no_policy_resource` in next rule that uses it

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: bundle/compliance/policy/gcp/iam/ensure_policy_not_managed_by_user.rego [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package compliance.policy.gcp.iam.ensure_policy_not_managed_by_user
+
+import data.compliance.policy.gcp.data_adapter
+import future.keywords.every
+import future.keywords.if
+
+default is_policy_not_managed_by_user = false
```
Comment (oren-zohar) [inline bundle/compliance/policy/gcp/iam/ensure_policy_not_managed_by_user.rego:7]: nit: double negation hurts readability in general IMO

**Review by oren-zohar — APPROVED** [review]

File: bundle/compliance/policy/gcp/iam/ensure_policy_not_managed_by_user.rego (hunk 3 above)
Comment (orouz) [reply bundle/compliance/policy/gcp/iam/ensure_policy_not_managed_by_user.rego:7]: agreed, but that's what the rule is for. adding a utility for `is_managed` and only using it with `not` seemed liked overkill. at least it's explicit

**Pull request merged**